data/scripts.ts

```diff
@@ -1182,6 +1182,11 @@ export const BattleScripts: BattleScriptsData = {
 				mustStruggle = false;
 			}
 			const move = this.dex.getMove(moveSlot.move);
+			// National Dex mechanic
+			if (move.gen > 7) {
+				zMoves.push(null);
+				continue;
+			}
```

Comment:
Do you have a link to an announcement thread or something, for this change?

Reply:
yes

https://www.smogon.com/forums/threads/national-dex-mechanics-thread.3656786/